tdrs-backend/tdpservice/users/authentication.py

```diff
@@ -8,9 +8,16 @@
 class CustomAuthentication(BaseAuthentication):
     """Define authentication and get user functions for custom authentication."""
 
-    def authenticate(self, username=None):
+    def authenticate(self, username=None, user_id=None):
         """Authenticate user with the request and username."""
         User = get_user_model()
+
+        if user_id:
+            try:
+                return User.objects.get(sub=user_id)
+            except User.DoesNotExist:
```

Comment:
Bypass the default authentication branch, and find a user by their unique `sub` `user_id` value.